enterprise/replicasync/replicasync.go

```diff
@@ -408,9 +408,6 @@ func (m *Manager) AllPrimary() []database.Replica {
 			continue
 		}
 
-		// When we assign the non-pointer to a
-		// variable it loses the reference.
-		replica := replica
```

Comment:
@johnstcn you called out this one as potentially being a concern, but I think there are already two "copies" happening here that I think make this safe

- it gets copied from the slice to the `replica` variable by the loop
- it gets copied from the `replica` variable to the argument value as part of the `append` call on the next line

ie. go does not _have_ "references", it has pointers and values. you're either copying a pointer or you're copying the whole value.

here's a go playground that hopefully instills confidence https://go.dev/play/p/MfFltxQepYE